pyron-core/src/main/scala/com/cloudentity/pyron/domain/Codecs.scala

```diff
@@ -136,8 +136,8 @@ object Codecs {
   implicit lazy val regexEnc: Encoder[Regex] = Encoder.encodeString.contramap(_.regex)
   implicit lazy val regexDec: Decoder[Regex] = Decoder.decodeString.map(_.r)
 
-  implicit lazy val PathMatchingEnc: Encoder[PathMatching] = deriveEncoder
-  implicit lazy val PathMatchingDec: Decoder[PathMatching] = deriveDecoder
+  implicit lazy val PathMatchingEnc: Encoder[PreparedRewrite] = deriveEncoder
```

Comment:
PreparedRewrite is bit different approach than PathMatching was
It has safe construction, which immediately checks, that regex is valid.
The way it is constructed it knows which path param is which capture group number, and always uses that for applying rewrites. This we can use same approach for processing both named {someParam} and numeric {1}, {-1} etc. way of referencing matching sections - by either name or position. Negative count from right. For usage in plugins it keeps param to allow param name for any capture-group number -> if it's named it returns name given in config, otherwise the number as string. So it does quite a bit during construction, and less is done in runtime to apply the rewrite. Also, it doesn't really use named regex-groups underneath, just works by group index. This allows using param naming not forced to be accepted regex group name, which would exclude names like {1} or {-1} or anything with underscore, which is now allowed {some_param_name}. Once it is applied to some incoming path in the request, AppliedRewrite is created with replacements applied. AppliedRewrite keeps reference to PreparedRewrite that was used, so we have complete picture.